x-pack/plugins/security/public/authentication/login/login_page.tsx

```diff
@@ -144,68 +175,120 @@ export class LoginPage extends Component<Props, State> {
-          <DisabledLoginForm
-            title={
-              <FormattedMessage
-                id="xpack.security.loginPage.unknownLayoutTitle"
-                defaultMessage="Unsupported login form layout."
-              />
-            }
-            message={
-              <FormattedMessage
-                id="xpack.security.loginPage.unknownLayoutMessage"
-                defaultMessage="Refer to the Kibana logs for more details and refresh to try again."
-              />
-            }
-          />
-        );
+    if (layout === 'error-es-unavailable') {
+      return (
+        <DisabledLoginForm
+          title={
+            <FormattedMessage
+              id="xpack.security.loginPage.esUnavailableTitle"
+              defaultMessage="Cannot connect to the Elasticsearch cluster"
+            />
+          }
+          message={
+            <FormattedMessage
+              id="xpack.security.loginPage.esUnavailableMessage"
+              defaultMessage="See the Kibana logs for details and try reloading the page."
+            />
+          }
+        />
+      );
+    }
+
+    if (layout === 'error-xpack-unavailable') {
+      return (
+        <DisabledLoginForm
+          title={
+            <FormattedMessage
+              id="xpack.security.loginPage.xpackUnavailableTitle"
+              defaultMessage="Cannot connect to the Elasticsearch cluster currently configured for Kibana."
+            />
+          }
+          message={
+            <FormattedMessage
+              id="xpack.security.loginPage.xpackUnavailableMessage"
+              defaultMessage="To use the full set of free features in this distribution of Kibana, please update Elasticsearch to the default distribution."
+            />
+          }
+        />
+      );
+    }
+
+    if (layout !== 'form') {
+      return (
+        <DisabledLoginForm
+          title={
+            <FormattedMessage
+              id="xpack.security.loginPage.unknownLayoutTitle"
+              defaultMessage="Unsupported login form layout."
+            />
+          }
+          message={
+            <FormattedMessage
+              id="xpack.security.loginPage.unknownLayoutMessage"
+              defaultMessage="Refer to the Kibana logs for more details and refresh to try again."
+            />
+          }
+        />
+      );
+    }
+
+    const loginSelector =
+      showLoginSelector &&
+      selector.providers.map((provider, index) => (
+        <EuiButton
+          key={index}
+          className="loginWelcome__selectorButton"
+          iconType="user"
+          fullWidth={true}
+          onClick={() => this.login(provider.type, provider.name)}
+        >
+          {provider.options.description ?? `${provider.type}/${provider.name}`}
+        </EuiButton>
+      ));
+
+    const loginSelectorAndLoginFormSeparator = showLoginSelector && showLoginForm && (
+      <>
+        <EuiText textAlign="center" color="subdued">
+          ―――&nbsp;&nbsp;
```

Comment:
__note:__ didn't have chance to find a better solution for this nasty `―――&nbsp;&nbsp;` temporary workaround yet, thinking...